addons/mail/models/mail_message.py

```diff
@@ -224,9 +235,9 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
             return super(Message, self)._search(
                 args, offset=offset, limit=limit, order=order,
                 count=count, access_rights_uid=access_rights_uid)
-        # Non-employee see only messages with a subtype (aka, no internal logs)
+        # Non-employee see only messages with a subtype and not internal
         if not self.env['res.users'].has_group('base.group_user'):
-            args = ['&', '&', ('subtype_id', '!=', False), ('subtype_id.internal', '=', False)] + list(args)
+            args = ['&', '&', ('subtype_id', '!=', False), ('is_internal', '=', False)] + list(args)
```

Comment:
Shouldn't this be using `_get_search_domain_share` for consistency?

Reply:
Even better, yes.